### PR TITLE
Add News Evaluation

### PR DESCRIPTION
**Pull request by ItayDev**
Change module names to lower case. Renamed `HypothesisModel` to `InferenceModel`. Added simple caching and prefetching for news.
Added logic for evaluating tweets with news articles

### CODE REVIEW

File: API.py [hunk 1]
```diff
@@ -1,18 +1,15 @@
...
-def run_hypothesis_model(body):
-    prediction = hypothesis_model.predict(body)
-    return {"predictions": prediction}
+from data.NewsRequester import NewsRequester
+from models.pattern import Parameters, DatasetPrepare, SignificantResultsFilter
+from models.inference.TweetsEvaluator import TweetsEvaluator
+from datetime import datetime
 
+inference_model = ModelLoader.load_inference_model()
+pattern_model = None # ModelLoader.load_pattern_model()
```
Comment (ShacharOr100) [inline API.py]: Why is pattern model not initialized?

File: data/NewsRequester.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+UTC = pytz.utc
+ARTICLE_DATE_FORMAT = "%Y-%m-%dT%H:%M:%S%z"
+QUERY_DATE_FORMAT = "%Y-%m-%d"
+API_KEY = "mkmf1ZFfJzxR1gmt2GUW0JppZXKDHg9M"
+API_URL = "https://api.nytimes.com/svc/search/v2/articlesearch.json?fq="
+NEWS_CACHE_PATH = "news_cache"
+NEWS_CACHE_FILE = "cache.pkl"
+
+
+class NewsRequester:
```
Comment (ShacharOr100) [inline data/NewsRequester.py:16]: I like this class much better than newsController

File: models/inference/TweetsEvaluator.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        articles_end_date = tweet_date + timedelta(days=ARTICLE_PREFETCH_AMOUNT)
+
+        articles = self.news_requester.request_news(articles_start_date, articles_end_date)
+
+        articles_with_scores = list(map(lambda article: {
+            'article': article,
+            'score': self.model.predict([{'premise': article['abstract'], 'hypothesis': tweet_content}])[0]
+        }, articles))
+
+        entailments = list(filter(lambda article: article['score'] > 0.7, articles_with_scores))
```
Comment (OrezriceWasHere) [inline models/inference/TweetsEvaluator.py:34]: I used SignificantResultHandler exactly for this purpose of filtering "interesting" tweets. Isn't better to let the dedicated class perform the operation?

**Review by ShacharOr100 — COMMENTED** [review]
Your code is well organized and easy to understand. I do like your work.

File: API.py (hunk 1 above)
Comment (ItayDev) [reply API.py]: Sorry it was for debugging and I forgot to return it

**Pull request merged**